app/models/items/recurring_donation.rb

```diff
@@ -6,4 +6,34 @@ class RecurringDonation < Item
 
   def one_line_description ; end
   def description_for_audit_txn ; end
+  def monthly_amount
+    # Finds the first donation with the recurring_donation_id matching this instance's id
+    donation = Donation.find_by(recurring_donation_id: id)
+    donation ? donation.amount : 0  # Returns the donation amount if found, otherwise returns 0
+  end
+  def first_donation
+    # Retrieves the earliest donation based on the created_at timestamp
+    Donation.where(recurring_donation_id: id).order(sold_on: :asc).first
+  end
+  def latest_donation
+    # Retrieves the most recent donation based on the created_at timestamp
+    Donation.where(recurring_donation_id: id).order(sold_on: :desc).first
+  end  
+  def total_to_date
```

Comment:
am i trippin or can't you just do this with RecurringDonation.find(id).donations.sum(:amount) or self.donations.sum(:amount) or smt like that? idk what the ruby equivalent of python's self is, but i just tried RecurringDonation.find(id).donations.sum(:amount) in the rails console and it seems to work. is there a reason you did it this way instead?

Reply:
Good point, that is a smarter way to go

Reply:
fixed